src/main/java/com/nvs/store/controllers/ProductController.java

```diff
@@ -30,6 +28,10 @@ public ResponseEntity<Product> getProduct(@PathVariable Long id) {
     public ResponseEntity<Product> addProduct(@RequestBody Product product) {
         return productService.addProduct(product);
     }
+    @PutMapping("{id}")
```

Comment:
please extract all ResponseEntities here from productService as as scope of conversion to response object should happen in controllers

Reply:
done